src/common/mockserver/__init__.py

```diff
@@ -35,27 +38,47 @@ def parse_request(self) -> bool:
             return False
         self.path, query = urlsplit(self.path)[2:4]
         self.query_params = parse_qs(query)
+        self.timestamp = time.time()
         logger.info(
             f"{self.command} request path: {self.path}, query params: {self.query_params}")
         return True
 
     def do_GET(self):
-        self.callback(Request(self.path, self.query_params))
-
-        if self.path.startswith("/report") or self.path.startswith("/debug") or self.path.startswith('/favicon'):
-            pass
+        self.callback(Request(self.path, self.query_params, self.timestamp))
+        response = self.response_provider(self.path, self.query_params)
+        response: Union[None, str, Tuple[int, Iterator[Tuple[str, str]], Union[None, str, bytes]]]
```

Comment:
This is not very readable. Some type aliases (made locally) could help:
```
StatusCode = int
Header = Tuple[str, str]
Body = Union[None, str, bytes]
Response = Tuple[StatusCode, Iterator[Header], Body]
```
and then
```response: Union[None, Body, Response]```
is much cleaner. And it wouldn't even be necessary if we had type hints in the initializer:
```
RequestPath = str
QueryParams = Dict[str, List[str]]
ProvidedResponse = Union[None, Body, Response]
callback: Callable[[Request], None]
reposnse_provider: Callable[[RequestPath, QueryParams], ProvidedResponse]
```